ts/src/pro/bitvavo.ts

```diff
@@ -506,6 +531,512 @@ export default class bitvavo extends bitvavoRest {
+    }
+
+    handleFetchBalance (client: Client, message) {
+        //
+        //    {
+        //        action: 'privateGetBalance',
+        //        response: [{
+        //                symbol: 'ADA',
+        //                available: '0',
+        //                inOrder: '0'
+        //            },
+        //            ...
+        //        ]
+        //    }
+        //
+        const action = this.safeString (message, 'action', 'privateGetBalance');
+        const messageHash = this.buildMessageHash (action, message);
+        const response = this.safeValue (message, 'response', []);
+        const balance = this.parseBalance (response);
+        client.resolve (balance, messageHash);
+    }
+
+    handleSingleOrder (client: Client, message) {
+        //
+        //    {
+        //        action: 'privateCreateOrder',
+        //        response: {
+        //            orderId: 'd71df826-1130-478a-8741-d219128675b0',
+        //            market: 'BTC-EUR',
+        //            created: 1689792749748,
+        //            updated: 1689792749748,
+        //            status: 'new',
+        //            side: 'sell',
+        //            orderType: 'limit',
+        //            amount: '0.0002',
+        //            amountRemaining: '0.0002',
+        //            price: '37000',
+        //            onHold: '0.0002',
+        //            onHoldCurrency: 'BTC',
+        //            filledAmount: '0',
+        //            filledAmountQuote: '0',
+        //            feePaid: '0',
+        //            feeCurrency: 'EUR',
+        //            fills: [],
+        //            selfTradePrevention: 'decrementAndCancel',
+        //            visible: true,
+        //            timeInForce: 'GTC',
+        //            postOnly: false
+        //        }
+        //    }
+        //
+        const action = this.safeString (message, 'action');
+        const response = this.safeValue (message, 'response', {});
+        const order = this.parseOrder (response);
+        const messageHash = this.buildMessageHash (action, response);
+        client.resolve (order, messageHash);
+    }
+
+    buildMessageHash (action, params = {}) {
+        const methods = {
+            'privateCreateOrder': this.actionAndMarketMessageHash,
+            'privateUpdateOrder': this.actionAndOrderIdMessageHash,
+            'privateCancelOrder': this.actionAndOrderIdMessageHash,
+            'privateGetTrades': this.actionAndMarketMessageHash,
+        };
+        const method = this.safeValue (methods, action);
+        let messageHash = action;
+        if (method !== undefined) {
+            messageHash = method.call (this, action, params);
+        }
+        return messageHash;
+    }
+
+    checkMessageHashDoesNotExist (messageHash) {
+        const supressMultipleWsRequestsError = this.safeValue (this.options, 'supressMultipleWsRequestsError', false);
```

Comment:
@pcriadoperez not sure why this is needed, can't we request the same thing twice and get the same result twice?

Reply:
I think it's needed because for some requests like `createOrderWs` we can't pass an id, so the the messageHash is just `privateCreateOrder`+marketId. So if for example someone calls createOrderWs twice with the same market before getting a response back, both requests would resolve on the first createOrder response which could lead to incorrect information. This warning is to prevent that.

I thought of making it more unique by including other parameters but if there is an error that's the identifiable information we get.

Reply:
@pcriadoperez So basically you're saying that we can't place several orders simultaneously using the WS channel?

Reply:
@pcriadoperez ? ⬆️  👀

Reply:
@carlosmiei , correct, for each order we have to wait for the acknolodegement it's been placed.
Only alternative solutions I can think of would be to use more clients equal to the numer of simultaneous orders.